niceman/distributions/redhat.py

```diff
@@ -0,0 +1,339 @@
+        the passed package (other)"""
+        if not isinstance(other, Package):
+            raise TypeError('satisfies() requires a package argument')
+        if not isinstance(other, RPMPackage):
+            return False
+        if self.name != other.name:
+            return False
+        if other.version is not None and self.version != other.version:
+            return False
+        if other.architecture is not None \
+                and self.architecture != other.architecture:
+            return False
+        return True
+
+_register_with_representer(RPMPackage)
+
+
+@attr.s
+class RedhatDistribution(Distribution):
+    """
+    Class to provide Redhat-based shell commands.
+    """
+
+    sources = TypedList(RPMSource)
+    packages = TypedList(RPMPackage)
+    version = attr.ib(default=None)  # version as depicted by /etc/redhat_version
+
+    def initiate(self, session):
+        """
+        Perform any initialization commands needed in the environment environment.
+
+        Parameters
+        ----------
+        environment : object
+            The Environment sub-class object.
+        """
+        lgr.debug("Adding Redhat update to environment command list.")
+        self._init_repo_sources(session)
+
+    def _init_repo_sources(self, session):
+        """
+        Install package repos listed in the spec but missing in the resource.
+
+        Parameters
+        ----------
+        session : Session object
+        """
+
+        # TODO: Expand the listing of 3rd party repos as popular ones in the
+        # neuroscience community become apparent.
+        third_party_repos = {
+            'epel/x86_64': 'epel-release'
+        }
+
+        # Get a list of the enabled repos on the system.
+        out, _ = session.execute_command(['yum', 'repolist'])
+        system_repos = []
+        skip = True
+        for line in out.splitlines():
+            if line.startswith('repo id'):
+                skip = False
+                continue
+            if skip:
+                continue
+            system_repos.append(line.split()[0])
+
+        # Try to install any repos listed in the spec that are missing on the system.
+        for source in self.sources:
+            if source.id not in system_repos:
+                if source.id in third_party_repos:
+                    session.execute_command(['yum', 'install', '-y',
+                        third_party_repos[source.id]])
+                else:
+                    lgr.error("Unable to install source repo \"%s\" found in \
+spec but not in resource environment.", source.id)
+
+    def install_packages(self, session, use_version=True):
+        """
+        Install the packages associated to this distribution by the provenance
+        into the environment.
+
+        Parameters
+        ----------
+        session : object
+        use_version : bool, optional
+          Use version information if provided.
+          TODO: support outside or deprecate
```

Comment:
what is that "deprecate" in this context?

Reply:
I meant to strike that from the code and forgot to go back and do it. This was originally in the DebianDistribution code so Butch can answer this question.
I was missing the implementation of the "user_version" parameter, so added that into the method